packages/mui-joy/src/Paper/Paper.tsx

```diff
@@ -0,0 +1,168 @@
+import { alpha } from '@mui/system';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import clsx from 'clsx';
+import { Mode } from 'packages/mui-system/src/cssVars/useCurrentColorScheme';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { useColorScheme, useThemeProps } from '../styles';
+import styled from '../styles/styled';
+import { getPaperUtilityClass } from './paperClasses';
+import { PaperProps, PaperTypeMap } from './PaperProps';
+
+// Originally from Material Paper
+// Tweaked to be used for Joy Paper
+const getOverlayAlpha = (elevationProp: 'xs' | 'sm' | 'md' | 'lg' | 'xl') => {
+  const elevationMap = {
+    xs: 5,
+    sm: 10,
+    md: 15,
+    lg: 20,
+    xl: 25,
+  };
+
+  const elevation = elevationMap[elevationProp];
+  let alphaValue;
+  switch (elevationProp) {
+    case 'xs':
+      alphaValue = 4.5 * Math.log(elevation) + 2;
+      break;
+    case 'sm':
+      alphaValue = 4.5 * Math.log(elevation) + 2;
+      break;
+    case 'md':
+      alphaValue = 4.5 * Math.log(elevation) + 2;
+      break;
+    case 'lg':
+      alphaValue = 4.5 * Math.log(elevation) + 2;
+      break;
+    case 'xl':
+      alphaValue = 4.5 * Math.log(elevation) + 2;
+      break;
+    default:
+      alphaValue = 4.5 * Math.log(elevation) + 2; // should be never reached
+      break;
+  }
+
+  return Number((alphaValue / 100).toFixed(2));
+};
+
+const useUtilityClasses = (ownerState: PaperProps) => {
+  const { elevation, variant, color, classes } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      variant && `variant${capitalize(variant)}`,
+      color && `color${capitalize(color)}`,
+      elevation && `elevation${capitalize(elevation)}`,
+    ],
+  };
+
+  return composeClasses(slots, getPaperUtilityClass, classes);
+};
+
+const PaperRoot = styled('div', {
+  name: 'MuiPaper',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      styles[ownerState.variant],
+      ownerState.variant === 'elevation' && styles[`elevation${ownerState.elevation}`],
+    ];
+  },
+})<{ ownerState: PaperProps & { mode?: Mode } }>(({ theme, ownerState }) => {
+  return [
+    {
+      // TODO: discuss the theme transition.
+      // This value is copied from mui-material Paper.
+      transition: 'box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+      ...(ownerState.elevation && {
+        boxShadow: theme.shadow[ownerState.elevation],
+        ...(ownerState.mode === 'dark' && {
+          backgroundImage: `linear-gradient(${alpha(
+            '#fff',
+            getOverlayAlpha(ownerState.elevation),
+          )}, ${alpha('#fff', getOverlayAlpha(ownerState.elevation))})`,
+        }),
+      }),
+      ...(!ownerState.variant &&
+        ownerState.color && {
+          backgroundColor: theme.vars.palette[ownerState.color]?.textColor,
+        }),
```

Comment:
```suggestion
```

It seems like we also don't need backgroundColor at initial.

Reply:
`variant` is undefined by default. I think this line is needed. Otherwise, passing `color` prop doesn't work in effect.

Reply:
I see but I don't think the background color should be palette.textColor, instead it should be white.

I think it is okay to add backgroundColor as default and let the variant overrides it.

```suggestion
      backgroundColor: theme.vars.palette.background.body,
```

Reply:
Sounds great!

Reply:
- I suggest we have the default value of `variant` to be `contained`.
- If we set it to `text` (as we discussed in Slack) by default, when only `color` prop is passed, text color will change, which is intended, but the background color doesn't change.
- I put the effects of combinations of props in the demo: https://deploy-preview-31124--material-ui.netlify.app/experiments/joy/paper/

Reply:
> I suggest we have the default value of variant to be contained

Can we try with `text` as the default variant first? My point is that this component would appear with `text` variant the most and the variant `contained | light` will appear on a few locations on a page.

> If we set it to text (as we discussed in Slack) by default, when only color prop is passed, text color will change, which is intended, but the background color doesn't change.

Yes, that's what I expect because the default variant is `text` which is more common for Sheet to have than `contained | light`.

Reply:
I'm good too with going with `text` as a default.